### DISCUSSION TITLE
The 'Switch Network' option does not work when connecting via a mobile wallet on a desktop device

### DISCUSSION THREAD

**Discussion by amandeep-coolplugin** [family/connectkit#301] [Q&A]
I tested your demo site, but it didn't send any network switch request on mobile. It only showed the network switched option on the desktop app, but in reality, it didn't function as expected.
https://docs.family.co/connectkit/try-it-out

**Comment by lochie** [comment]
@amandeep-coolplugin what wallet were you using? could you share an example (e.g screen recording) with what occurred?

**Reply by lochie** [reply]
Worth noting that wallets utilising WalletConnect 2.0 don't require a prompt when switching network as all chains are approved when connecting your wallet, so switching networks on the client side won't needing approval within the wallet. This may be the confusion here.

**Reply by amandeep-coolplugin** [reply]
I'm using the Metamask Wallet on an Android mobile device. In my case, the current active network on my mobile is BNB mainnet. However, on your site, only some specific blockchain networks are supported. When I attempt to switch the network, a prompt appears, but clicking on any network option doesn't lead to any action on the mobile device.

Please inform me if Metamask Wallet does not support WalletConnect 2.0 ?

**Reply by lochie** [reply]
Does tapping a different network in the ConnectKit prompt allow you to close the modal? If so then that means the switch was successful as WalletConnect 2.0 doesn't require confirmation within the wallet.

If you can share a screen recording this might help us debug

**Reply by amandeep-coolplugin** [reply]
I have recorded a video demonstrating the issue:

[Video Recording](https://www.awesomescreenshot.com/video/20591213?key=86be9ec45b5d4b2a4ca0d32b05120ca7)

Additionally, you can access the video on Google Drive:

[Google Drive Video](https://drive.google.com/file/d/1Njm5afEoSg9I-7eVyrY6qsmY5DOVJDKq/view?usp=sharing)

I have thoroughly tested this on various mobile devices, including iOS, and observed that while the network indicator on the site shows that it is switching, it doesn't actually change the network on the mobile device.

It's worth noting that I also tested a similar process using [Web3Modal](https://web3modal.com/), where the network switch request should ideally reach the mobile device and trigger a network change. However, I found that this method is not always reliable, as sometimes the request doesn't reach the device.

Please review the video and the issue description for further investigation into this network switching problem.
